code/weapon/weapons.cpp

```diff
@@ -1360,11 +1362,17 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 	}
 
 	if (optional_string("$Damage Multiplier over Lifetime Curve:")) {
+		//Legacy table. Just populates the modular curve set!
 		SCP_string curve_name;
 		stuff_string(curve_name, F_NAME);
-		wip->damage_curve_idx = curve_get_by_name(curve_name);
-		if (wip->damage_curve_idx < 0)
+		int curve = curve_get_by_name(curve_name);
+		if (curve < 0)
 			Warning(LOCATION, "Unrecognized damage curve '%s' for weapon %s", curve_name.c_str(), wip->name);
+
+		if (subtype == WP_BEAM)
+			wip->beam_hit_curves.add_curve("Lifetime", weapon_info::BeamHitCurveOutputs::DAMAGE_MULT, modular_curves_entry{ curve });
```

Comment:
This came up recently, most beams will not be `WP_BEAM` (indeed, no retail beams are). And yes, that is not known until we see the "Beam" weapon info flag of $BeamInfo...

Don't think there's anything else for it but to fill something temporary, and then populate it properly once it's known for sure.